llama_index/vector_stores/postgres.py

```diff
@@ -39,11 +59,19 @@ class AbstractData(base):  # type: ignore
 
 
 class PGVectorStore(VectorStore):
+    from sqlalchemy.sql.selectable import Select
+
     stores_text = True
     flat_metadata = False
 
     def __init__(
-        self, connection_string: str, async_connection_string: str, table_name: str
+        self,
+        connection_string: str,
+        async_connection_string: str,
+        table_name: str,
+        hybrid_search: bool = False,
```

Comment:
This is the flag to enable hybrid_search support in postgres. Note that for prior/existing databases this requires a schema migration to work (brand new databases will have the correct schema). Similar to how the pinecone implementation requires coordination from the user to be set up properly (i.e. the user needs to provide their own index with dot product similarity), users with existing databases would need have performed the necessary migration before enabling this feature. The alternative would be to perform schema introspection and perform the migration as needed although this functionality is not supported by sqlalchemy directly. Let me know your thoughts on how to proceed re: this api.

Reply:
@gitpeppy would existing users still need a DB migration if `hybrid_search` is set to False here?

If so, would it be possible to instead leave `AbstractData` as is and instead create a `class HybridAbstractData(AbstractData)` with the new fields instead? We could then just have the return value of `get_data_model` be the original class or the hybrid class based on the `hybrid_search` flag.

Reply:
@sourabhdesai thanks for the comments!

If an existing user has `hybrid_search` set to `False` then they wouldn't need a DB migration (and none would be performed). Do you have a sense of how important it is for hybrid support to be customizable? From a usability standpoint, I think the simplest/easiest to grok model would be if the `PGVectorStore` just always supported hybrid queries. The downside of this is the additional overhead of an unused `ts_vector` column if the user is only storing embeddings. On the plus side, we can avoid the somewhat weird interface now of a partially supported "hybrid_query" API that can throw exceptions at runtime. To support this would require proactively doing schema migrations though which seems to require another library (alembic?).

I like your idea of splitting out into two separate classes! I'll make that change (I think I was initially freaked out by how subclassing base_declarations adds metadata so tried to stuff everything into the existing class)